### DISCUSSION TITLE
Unable to make Seq available under a different path

### DISCUSSION THREAD

**Discussion by 1234** [datalust/seq-tickets#1327] [Help]
By default Seq is listing to http://localhost/ and http://localhost:5341/
`Seq listening on ["http://localhost/", "http://localhost:5341/"]`

I'm trying to run Seq for specific environment which is available under the domain `example-domain-dev.corporate-network.local`.
Because a another service is already listening there, I try to make the Seq service available under a different patch (like `example-domain-dev.corporate-network.local/seq`)
I have adjusted the following variable for the change: 
`SEQ_API_LISTENURIS=http://localhost/seq,http://localhost:5341/`

Unfortunately, the following error occurs after that:
```
────────────────────────────────────────
  Seq ♦ Machine data, for humans.
─────────── © 2021 Datalust Pty Ltd ────

Running as server; press Ctrl+C to exit.

[10:04:25 INF] Seq "2021.2.6459" running on OS "Linux 4.18.0-305.12.1.el8_4.x86_64 #1 SMP Mon Jul 26 08:06:24 EDT 2021"
[10:04:25 INF] Seq detected 67240.914944 MB of RAM
[10:04:25 INF] Opening event store at "/data/Stream/stream.flare"
[10:04:25 INF] Opening metastore "/data/Documents/documents.lmdb"
[10:04:25 INF] Storage subsystem available
[10:04:25 FTL] Unable to start Kestrel.
System.InvalidOperationException: A path base can only be configured using IApplicationBuilder.UsePathBase().
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.ParseAddress(String address, Boolean& https)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.AddressesStrategy.BindAsync(AddressBindContext context)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.BindAsync(IEnumerable`1 listenOptions, AddressBindContext context)
   at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerImpl.BindAsync(CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerImpl.StartAsync[TContext](IHttpApplication`1 application, CancellationToken cancellationToken)
```

It is possible to pass the app.UsePathBase variable to the application or is there a better solution? (I didn't find anything in the documentation)

Seq is runing in docker with 2021.2 image and has a nginx as a reverse proxy in front of it.

**Comment by nblumhardt** [comment]
Hi @1234, thanks for dropping by.

Using Docker/Linux, you'll need to do the path mapping at the reverse proxy rather than at Seq itself.

You won't need to set `SEQ_API_LISTENURIS`, but will need to set `SEQ_API_CANONICALURI`:

```
SEQ_API_CANONICALURI=https://example-domain-dev.corporate-network.local/seq
```

You'll then need to have the location `/seq` forward to `/` on the container, in the nginx proxy configuration. I don't have an example ready at hand, but let me know if you need help tracking some information down on how to do this.

HTH!
Nick

**Comment by 1234** [comment]
Hello Nick,

thanks for the fast answer! 
I've already tried your suggestion at first and its unfortunately doesn't work for me. 

Here is the nginx side conf:
```
server{
  listen 443 ssl;
  server_name example-domain-dev.corporate-network.local;
  client_max_body_size 0;
 
  #seq
  location /seq/ {
    proxy_pass                          http://seq:80;
  }
}
```
My Seq env:
```
ACCEPT_EULA=Y
SEQ_API_CANONICALURI=https://example-domain-dev.corporate-network.local/seq
```

The same nginx side conf works for RabbitMQ Management without any problem. 

Following nginx side conf works for `example-domain-dev.corporate-network.local` without `/seq`. 
```
server{
  listen 443 ssl;
  server_name example-domain-dev.corporate-network.local;
  client_max_body_size 0;
 
  #seq
  location / {
    proxy_pass                          http://seq:80;
  }
}
```



Hope you can help me!

Kind regards
Alexey

**Reply by nblumhardt** [reply]
Hi Alexey,

Looks like you may need a trailing slash in the argument to `proxy_pass` in order for nginx to perform the rewrite correctly. There's a lot of discussion in:

https://serverfault.com/questions/379675/nginx-reverse-proxy-url-rewrite

with a few different approaches mentioned. Let me know if this helps!

Best regards,
Nick

**Reply by 1234** [reply]
Hi Nick, 

thank you a lot! 

works like it should: 
```
server{
  listen 443 ssl;
  server_name example-domain-dev.corporate-network.local;
 
  #seq
  location /seq/ {
    rewrite /seq/(.*) /$1  break;
    proxy_pass                          http://seq:80;
  }
}
```

Best regards,
Alexey
